src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -572,24 +571,31 @@ async Task OnStreamCanceled(InvocationRequest irq)
             }
 
             args = newArgs.ToArray();
-
-            return readers;
         }
 
-        private void LaunchStreams(Dictionary<string, object> readers, CancellationToken cancellationToken)
+        private void LaunchStreams(CancellationToken cancellationToken)
         {
-            if (readers == null)
+            if (_readers == null)
             {
                 // if there were no streaming parameters then readers is never initialized
                 return;
             }
-            foreach (var kvp in readers)
+            foreach (var kvp in _readers)
             {
                 var reader = kvp.Value;
 
                 // For each stream that needs to be sent, run a "send items" task in the background.
                 // This reads from the channel, attaches streamId, and sends to server.
                 // A single background thread here quickly gets messy.
+#if NETCOREAPP3_0
+                if (ReflectionHelper.IsIAsyncEnumerable(reader.GetType()))
+                {
+                    _ = _sendIAsyncStreamItemsMethod
```

Comment:
Long term, we should store the Takss returned by _sendIAsyncStreamItemsMethod and _sendStreamItemsMethod and track them during connection teardown and log warnings if they don't complete within a timeout.